src/app/dataset-block/metadata-block/components/event-details/builder.events.ts

```diff
@@ -40,15 +49,120 @@ type SetPollingSourceScalarSections =
     | SetPollingSourceSection.PREPROCESS
     | SetPollingSourceSection.READ;
 
-abstract class EventSectionBuilder<TEvent> {
-    public abstract buildEventSections(event: TEvent): EventSection[];
+enum SetTransformSection {
+    INPUTS = "inputs",
+    TRANSFORM = "transform",
 }
 
-class SetPollingSourceSectionBuilder extends EventSectionBuilder<SetPollingSource> {
+type SetTransformScalarSections =
+    | SetTransformSection.INPUTS
+    | SetTransformSection.TRANSFORM;
+
+export const descriptorMapper = {
+    [EventTypes.SetPollingSource]: SET_POLLING_SOURCE_DESCRIPTORS,
+    [EventTypes.SetTransform]: SET_TRANSFORM_SOURCE_DESCRIPTORS,
+};
+
+abstract class EventSectionBuilder<TEvent> {
     private readonly UNSUPPORTED_ROW_DESCRIPTOR: EventRowDescriptor = {
-        ...SET_POLLING_SOURCE_DESCRIPTORS[`SetPollingSource.UnsupportedKey`],
+        label: "",
+        tooltip: "Unsupported value",
+        presentationComponent: UnsupportedPropertyComponent,
+        separateRowForValue: false,
+        dataTestId: "unsupportedKey",
     };
+    public abstract buildEventSections(event: TEvent): EventSection[];
+    public buildEventRows(
+        event: SetPollingSource | SetTransform,
```

Comment:
I think we can use `TEvent`. If we are missing some base interface for all event kinds (i.ex., something that has a `__typename` and key/values), we can add one to ensure compilation passes.

Reply:
Yes, fixed onTEvent type